data/lua/diversion.lua

```diff
@@ -0,0 +1,95 @@
+
+local _ = wesnoth.textdomain 'wesnoth-help'
+local T = wml.tag
+local on_event = wesnoth.require("on_event")
+
+local u_pos_filter = function(u_id)
+
+        local output = "initial"
+        local hex_dirs = {"n", "ne", "se", "s", "sw", "nw"}
+        local diversion_unit = wesnoth.units.get(u_id)
+        if not diversion_unit then
+            return nil
+        end
+        for i, dir in ipairs(hex_dirs) do
+            if diversion_unit:matches {
+              id = u_id, 
+              T.filter_adjacent {
+                  is_enemy = "yes",
+                  adjacent = dir,
+                  formula = "self.hitpoints > 0",
+                  T.filter_adjacent {
+                      is_enemy = "yes",
+                      adjacent = dir,
+                      formula = "self.hitpoints > 0"
+                  }
+              }
+            } then
+                output = "diverter"
+                break 
+            end
+        end
+        if output ~= "initial" then
+            return output
+        else
+        -- either nothing passed filter, or there was an error
+            return nil
+        end
+end
+
+
+local status_anim_update = function()
+
+	local ec = wesnoth.current.event_context
+
+	if not ec.x1 or not ec.y1 then
+                return
+        end
+
+        -- find all units on map with ability = diversion but not status.diversion = true
+        local div_candidates = wesnoth.units.find_on_map({
+                ability = "diversion",
+                {"not", { status = "diversion" }}
+                })
+        -- for those that pass the filter now, change status and fire animation
+        for index, ec_unit in ipairs(div_candidates) do
+                local filter_result = u_pos_filter(ec_unit.id)
+                if filter_result then
+                    ec_unit.status.diversion = true
+                    ec_unit:extract()
```

Comment:
whats the reason for calling extract() foillowed by a to_map() here=

Reply:
Not sure, I was just following what was in the petrify/unpetrify tag.